src/js/models/TableColumns.js

```diff
@@ -16,21 +16,23 @@ export default class TableColumns {
     columns: Column[] = [],
     tableSetting: ColumnSettingsMap = {}
   ) {
+    // $FlowFixMe
+    const defaultVisible = Object.values(tableSetting).every((c) => c.isVisible)
```

Comment:
I had tried this, but if you open the column chooser and deselect an option then it will deselect all of them since this will return false for every one

Reply:
Thanks for the heads up @mason-fish. You were correct. The new commits fix this by immediately adding any columns from a search results to the "Column Preferences" state with sane defaults. Then the "select all" and deselect all" operate on all columns known to the preferences. Not just the columns that are in the current search results. Hope that makes sense.